share/p2p/discovery/discovery.go

```diff
@@ -225,8 +238,8 @@ func (d *Discovery) discoveryLoop(ctx context.Context) {
 					logInterval, d.set.Size(), d.set.Limit(),
 				)
 			}
-			// Do not break the loop; just continue
-			continue
+		case <-d.triggerDisc: // Listen for triggerDisc channel
+			runDiscovery()
```

Comment:
This would cause discovery to retry without respect of backoff. At this point we know that we need to trigger discovery, since the loop has started, there is no need to skip backoff in this case.